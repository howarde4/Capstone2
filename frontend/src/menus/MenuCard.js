import React from "react";
import { Link } from "react-router-dom";
import './Menu.css';

// Displays link to specific menus (MenuDetail). MenuList -> MenuCard

function MenuCard({ username, menu_event, id }) {
  console.debug("MenuCard");

  return (
    <div className='linkDiv'>
      <Link to={`/menus/${username}/${id}`}>
        <button className='MenusLink'>{menu_event}</button>
      </Link>
    </div>
);
}

export default MenuCard;
