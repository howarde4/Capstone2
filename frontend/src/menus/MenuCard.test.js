import React from "react";
import { render } from "@testing-library/react";
import MenuCard from "./MenuCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
    const event = "Test Event";
    const user = "testUser1";
    const id = 1;
    const key = 1;

  const { asFragment } = render(
      <MemoryRouter>
        <MenuCard menu_event={event} username={user} id={id} key={key} />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

