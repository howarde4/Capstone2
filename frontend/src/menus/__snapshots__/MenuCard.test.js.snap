// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="linkDiv"
  >
    <a
      href="/menus/testUser1/1"
    >
      <button
        class="MenusLink"
      >
        Test Event
      </button>
    </a>
  </div>
</DocumentFragment>
`;
